src/state.ts

```diff
@@ -38,7 +46,7 @@ export async function refreshPiggybankState(rpc: ConcordiumGRPCClient, contract:
             );
         }
         case 'success': {
-            const buffer = toBuffer(result.returnValue || '', 'hex');
+            const buffer = Buffer.from(ReturnValue.toBuffer(result.returnValue ?? ReturnValue.empty()));
```

Comment:
I wonder if this is the correct way to handle an empty result... That should never happen right?  I understand that you're just migrating the existing behavior, but consider improving error handling while you're at it. You can also just add a comment that error handling is lacking (or sufficient)...